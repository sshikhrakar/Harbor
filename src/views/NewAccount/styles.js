import { StyleSheet } from 'react-native';

import fonts from '../../config/fonts';
import colors from '../../config/colors';

const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    paddingVertical: 20,
    paddingHorizontal: 24,
    backgroundColor: colors.SILVER,
  },

  logoContainer: {
    flex: 1,
    paddingVertical: 28,
    paddingHorizontal: 18,
    alignItems: 'center',
    justifyContent: 'center',
  },

  landingPageLogo: {
    flex: 1,
    resizeMode: 'contain',
  },

  greetingContainer: {
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  subText: {
    fontSize: 18,
    fontFamily: fonts.primary.LIGHT,
  },

  error: {
    fontSize: 14,
    alignSelf: 'center',
    textAlign: 'center',
    color: colors.ERROR,
    fontFamily: fonts.primary.LIGHT,
  },

  emailPasswordContainer: {
    flex: 1,
    justifyContent: 'space-around',
  },

  textInputField: {
    height: 44,
    borderWidth: 0.8,
    borderRadius: 6,
    paddingHorizontal: 10,
    borderColor: colors.ASBESTOS,
    fontFamily: fonts.primary.REGULAR,
  },

  modalButtonsContainer: {
    flex: 1,
  },

  button: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.ASBESTOS,
  },

  cancelButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.CRIMSON_RED,
  },


});

export default styles;
