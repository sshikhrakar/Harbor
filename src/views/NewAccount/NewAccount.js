import React from 'react';
import {
  View,
  Text,
  Image,
  TextInput,
} from 'react-native';
import PropTypes from 'prop-types';

import { ButtonWithSpinner } from '../../components';

import styles from './styles';
import colors from '../../config/colors';
import images from '../../config/images';

function NewAccount(props) {

  const {
    onSubmit,
    onCancel,
    emailText,
    isSigningUp,
    passwordText,
    setEmailText,
    setPasswordText,
  } = props;

  return (
    <View style={ styles.modalContainer }>

      <View style={ styles.logoContainer}>
        <Image
          style={ styles.landingPageLogo }
          source={ images.harborLogo }
          tintColor={ colors.GREY }
        />
      </View>

      <View style={ styles.greetingContainer}>
        <Text style={ styles.subText }>
          CREATE AN ACCOUNT
        </Text>
      </View>

      <View style={ styles.emailPasswordContainer }>
        <TextInput
          value={ emailText }
          placeholder={ 'E-mail' }
          onChangeText={ setEmailText }
          style={ styles.textInputField }
          keyboardType="email-address"
          underlineColorAndroid={ 'transparent' }
        />
        <TextInput
          value={ passwordText }
          secureTextEntry= { true }
          placeholder={ 'Password' }
          style={ styles.textInputField }
          onChangeText={ setPasswordText }
          underlineColorAndroid={ 'transparent' }
        />
      </View>

      <View style={ styles.modalButtonsContainer }>
        <ButtonWithSpinner
          buttonStyle={ styles.button }
          isLoading={ isSigningUp }
          text="CREATE"
          onPress={ onSubmit }
        />

        <ButtonWithSpinner
          buttonStyle={ styles.cancelButton }
          isLoading={ false }
          text="CANCEL"
          onPress={ onCancel }
        />
      </View>

    </View>
  );

}

NewAccount.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  emailText: PropTypes.string.isRequired,
  isSigningUp: PropTypes.bool.isRequired,
  passwordText: PropTypes.string.isRequired,
  setEmailText: PropTypes.func.isRequired,
  setPasswordText: PropTypes.func.isRequired,
};

export default NewAccount;
