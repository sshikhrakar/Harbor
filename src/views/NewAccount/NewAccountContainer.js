import {
  compose,
  withState,
} from 'recompose';
import { connect } from 'react-redux';

import NewAccount from './NewAccount';

const mapStateToProps = (state) => ({
  
});

const mapDispatchToProps = (dispatch) => ({
  
});

export const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),

  withState(),
);

export default enhance(NewAccount);
