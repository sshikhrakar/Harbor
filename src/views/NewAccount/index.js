import NewAccountContainer from './NewAccountContainer';

export default NewAccountContainer;
