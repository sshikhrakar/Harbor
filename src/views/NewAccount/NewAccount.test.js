import React from 'react';
import renderer from 'react-test-renderer';

import NewAccount from './NewAccount';
import { enhance } from './NewAccountContainer';

describe('VIEWS: NewAccount', () => {

  it('should render correctly', () => {
    const tree = renderer.create(enhance(<NewAccount />)).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
