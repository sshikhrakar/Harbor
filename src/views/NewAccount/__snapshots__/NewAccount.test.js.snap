// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VIEWS: NewAccount should render correctly 1`] = `null`;
